### ISSUE TITLE
vmap collectives on constants

### ISSUE THREAD

**Issue by mattjj** [google/jax#5515]
See #5509. Things like this work:

```python
import jax

jax.vmap(lambda x: lax.psum(1, 'i'), axis_name='i')(jnp.arange(3))
jax.jit(jax.vmap(lambda x: lax.psum(1, 'i'), axis_name='i')(jnp.arange(3))
```

But not this:

```python
jax.vmap(lambda x: lax.pmin(1, 'i'), axis_name='i')(jnp.arange(3))
jax.jit(jax.vmap(lambda x: lax.pmin(1, 'i'), axis_name='i')(jnp.arange(3))
```

The first line doesn't work because pmin doesn't have an impl rule, and even if it did the second line wouldn't work because the `vmap` trace would miss it and then the axis would be unbound when we do the `jit` lowering.

We solve the second issue for `psum` with [a special `bind` rule](https://github.com/google/jax/blob/84f55f87ecd4b1a69ed7eac48b144b57df9d3f11/jax/_src/lax/parallel.py#L485-L488). We should do that more generally for collectives.

**Comment by jekbradbury** [comment]
I do this in the traceable in avals-with-names, but then it's missed when we eval_jaxpr (or the equivalent) inside a vmap—something that seems not to be tested in the codebase now but should probably work! I agree that a custom bind rule (or making vmap dynamic 😛 ) is better.

**Comment by rajasekharporeddy** [comment]
Hi @mattjj 

I tried to execute the mentioned code in Colab using JAX version 0.4.23 and observed that it executes without any issue or error.

```python
import jax
print(f"JAX version: {jax.__version__}")
```
Output: 
`JAX version: 0.4.23`

```python
import jax
from jax import lax
import jax.numpy as jnp

jax.vmap(lambda x: lax.psum(1, 'i'), axis_name='i')(jnp.arange(3))
```
Output: 
`Array([3, 3, 3], dtype=int32, weak_type=True)`

```python
jax.jit(jax.vmap(lambda x: lax.psum(1, 'i'), axis_name='i'))(jnp.arange(3))
```
Output: 
`Array([3, 3, 3], dtype=int32, weak_type=True)`

```python
jax.vmap(lambda x: lax.pmin(1, 'i'), axis_name='i')(jnp.arange(3))
```
Output: 
`Array([1, 1, 1], dtype=int32, weak_type=True)`

```python
jax.jit(jax.vmap(lambda x: lax.pmin(1, 'i'), axis_name='i'))(jnp.arange(3))
```
Output: 
`Array([1, 1, 1], dtype=int32, weak_type=True)`

Kindly find the [gist](https://colab.research.google.com/gist/rajasekharporeddy/ccd550b546f4fb3872ef2332010a77b6/-5515.ipynb) for reference.

Please let me know if my understanding of this issue is correct?

Thank you.

**Comment by mattjj** [comment]
Thanks, @rajasekharporeddy ! Indeed it looks like we fixed these. And I have no idea what I was talking about with the pmin stuff.

I really appreciate you testing these and pointing them out!
